### PR TITLE
Repair-NsxEdge: Add Upgrade feature to Nsx-Edge

### PR DESCRIPTION
**Pull request by alagoutte**
It is now possible to upgrade Edge with a cmdlet !

### CODE REVIEW

File: module/PowerNSX.psm1 [hunk 1]
```diff
@@ -13695,7 +13695,7 @@ function Repair-NsxEdge {
     up to 200 subinterfaces.  Multiple external IP addresses can be configured
     for load balancer, site‐to‐site VPN, and NAT services.
 
-    The Repair-NsxEdge cmdlet allows a Resync or Redploy operation to be
+    The Repair-NsxEdge cmdlet allows a Resync, Redploy or Upgrade operation to be
     performed on the specified Edges appliance.
```
Comment (nmbradford) [inline module/PowerNSX.psm1]: Edges -> Edge.

File: module/PowerNSX.psm1 [hunk 2]
```diff
@@ -13728,7 +13733,8 @@ function Repair-NsxEdge {
             #Specify the repair operation to be performed on the Edge.
             #If ForceSync - The edge appliance is rebooted
             #If Redeploy - The Edge is removed and redeployed (if the edge is HA this causes failover, otherwise, an outage.)
-            [ValidateSet("ForceSync", "Redeploy")]
+            #If Upgrade - The Edge is upgrade to last release
```
Comment (nmbradford) [inline module/PowerNSX.psm1]: change to -> upgraded to latest release

**Comment by nmbradford** [comment]
jenkins test this please

**Comment by alagoutte** [comment]
typo fixed !

**Comment by alagoutte** [comment]
Repair-NsxEdge is not very good name for upgrade..
may be add a alias ?

need also the same function for NsxLogicalRouter ... (there is no Repair-NsxLogicalRouter for the moment...)

**Comment by nmbradford** [comment]
Ok - did you want to raise another PR for this, or leave this unmerged until you have done that?

**Comment by alagoutte** [comment]
i will push a another PR later...

**Pull request merged**